File: server/features/interaction.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

// InteractionLog describes the log-user-interaction RPC.
var InteractionLog = featureregistry.Feature{
	ID:          "interaction-log",
	Title:       "Log User Interaction",
	Description: "Records a user interaction event associated with a session for analytics and auditing.",
	RPCIDs:      []string{"interaction:log"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(InteractionLog)
}

File: web-app/src/lib/features/features/interaction.ts
import type { Feature } from '../types';

export const interactionFeatures = {
  'interaction-log': {
    id: 'interaction-log',
    title: 'Log User Interaction',
    description: 'Records a user interaction event associated with a session for analytics and auditing.',
    rpcIds: ['interaction:log'],
    componentPaths: [],
    testIds: [
      'TestLogInteraction_Success',
      'TestLogInteraction_NoSessionID',
    ],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;
